plugins/node/opentelemetry-plugin-mongodb/src/mongodb.ts

```diff
@@ -192,13 +192,18 @@ export class MongoDBPlugin extends BasePlugin<typeof mongodb> {
     });
 
     if (command === undefined) return;
-    const query = Object.keys(command.query ?? command.q ?? command).reduce(
-      (obj, key) => {
-        obj[key] = '?';
-        return obj;
-      },
-      {} as { [key: string]: string }
-    );
+
+    // capture parameters within the query as well if enhancedDatabaseReporting is enabled.
+    const query = this._config?.enhancedDatabaseReporting
```

Comment:
can you compare this directly with boolean expression
`this._config?.enhancedDatabaseReporting === true`
simply to avoid situation like this `enhancedDatabaseReporting = 'true'` or `enhancedDatabaseReporting='foo'` etc.

Reply:
fixed this.